### PR TITLE
Add max_parallel parameter to MySQL backend.

### PR DESCRIPTION
**Pull request by ikatson**
This limits the number of concurrent connections, so that vault does not die suddenly from "Too many connections".

This can happen when e.g. vault starts up, and tries to load all the existing leases in parallel. At the time of writing this, the value [ExpirationRestoreWorkerCount](https://github.com/hashicorp/vault/blob/v0.7.2/helper/consts/consts.go#L6) is set to 64, meaning that if there are enough leases in the vault's DB, it will generate AT LEAST 64 concurrent connections to MySQL when loading the data during start-up. On certain configurations, e.g. smaller AWS RDS/Aurora instances, this will cause Vault to fail startup.

### CODE REVIEW

**Comment by jefferai** [comment]
Is there a reason not to do this with the permit pool and `max_parallel` like most other backends?

**Comment by ikatson** [comment]
@jefferai sure, didn't know about its existence. Reimplemented now with PermitTool and rebased.

Although it seems a little bit slower, probably the lock is coarser than internal db/mysql one.

**Comment by ikatson** [comment]
That CI error does not seem related, seems like smth else in master is broken

```
command/generate-root_test.go:85: ui.OutputWriter.Bytes undefined (type *cli.syncBuffer has no field or method Bytes)
command/rekey_test.go:185: ui.OutputWriter.Bytes undefined (type *cli.syncBuffer has no field or method Bytes)
command/write_test.go:246: ui.OutputWriter.Bytes undefined (type *cli.syncBuffer has no field or method Bytes)
command/write_test.go:247: ui.OutputWriter.Bytes undefined (type *cli.syncBuffer has no field or method Bytes)
```

File: website/source/docs/configuration/storage/mysql.html.md [hunk 1]
```diff
@@ -42,6 +42,9 @@ storage "mysql" {
 - `tls_ca_file` `(string: "")` – Specifies the path to the CA certificate to
   connect using TLS.
 
+- `max_parallel` `(string: "128")` – Specifies The maximum number of concurrent
```
Comment (jefferai) [inline website/source/docs/configuration/storage/mysql.html.md]: s/The/the

**Comment by ikatson** [comment]
@jefferai made the changes, ready for another iteration. I can rebase into one commit when you think it's good enough.

Thanks @jefferai

**Pull request merged**